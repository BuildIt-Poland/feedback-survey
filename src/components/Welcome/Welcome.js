import React, { useState, useEffect } from 'react';
import { Box, Heading, Grommet } from 'grommet';

import { spacingSmall, spacingLarge, fontXLarge, fontLarge } from '../../styles/designTokens';
import Agreed from '../Agreed';
import LinkButton from '../Button/LinkButton';
import { surveyMatchParams } from '../../types';

const theme = {
  heading: {
    weight: 500,
    level: {
      1: {
        medium: {
          size: fontXLarge
        }
      },
      2: {
        medium: {
          size: fontLarge
        }
      }
    }
  },
  paragraph: {
    extend: {
      'max-width': 'none'
    }
  },
  box: {
    extend: {
      background: 'url("/image/human_in_hand.png") no-repeat',
      'background-position-x': '1rem',
      'background-position-y': 'calc(100% - 3rem)'
    }
  }
};

const Welcome = ({ match = { params: {} } }) => {
  const [surveyId, setSurveyID] = useState('');
  const [clientName, setClientName] = useState('');
  const [path, setPath] = useState('');

  useEffect(() => {
    setSurveyID(match.params.surveyId ? match.params.surveyId : '');
    setClientName(match.params.clientName ? match.params.clientName : '');
    setPath(surveyId && clientName ? `/survey/${surveyId}/${clientName}` : '/survey');
  });

  return (
    <Grommet theme={theme}>
      <Box align="center" justify="center" pad="medium" height="100vh">
        <Heading level="1" margin={{ bottom: 'none' }}>
          Performance Snapshot on RSR Consultant:
        </Heading>
        <Heading level="2" margin={{ top: spacingSmall, bottom: spacingLarge }}>
          {clientName}
        </Heading>
        <Agreed />
        <LinkButton linkTo={path} label="start" />
      </Box>
    </Grommet>
  );
};

Welcome.propTypes = {
  match: surveyMatchParams
};

export default Welcome;
