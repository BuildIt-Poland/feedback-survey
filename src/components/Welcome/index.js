import { withRouter } from 'react-router-dom';

import Welcome from './Welcome';

export default withRouter(Welcome);
