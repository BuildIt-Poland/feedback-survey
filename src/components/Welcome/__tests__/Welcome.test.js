import React from 'react';
import { create } from 'react-test-renderer';

import Welcome from '../Welcome';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - Welcome', () => {
  it('renders correctly with params', () => {
    const match = {
      params: {
        surveyId: '12345',
        employeeName: 'John'
      }
    };
    const component = create(<Welcome match={match} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly without params', () => {
    const component = create(<Welcome />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
