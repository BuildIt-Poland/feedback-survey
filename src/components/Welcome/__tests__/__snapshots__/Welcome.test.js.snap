// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Welcome renders correctly with params 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <div
    className="StyledBox-sc-13pk1d4-0 lcYpOF"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 jIghTj"
    >
      Performance Snapshot on RSR Consultant:
    </h1>
    <h2
      className="StyledHeading-sc-1rdh4aw-0 fDHdzw"
    >
      John
    </h2>
    <section
      className="sc-bdVaJa gPpPXU"
    >
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        The feedback you provide will be used for individual development and career pathway and forms part of RSR Reward and Recognition scheme.
      </p>
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        Feedback will also be used to enhance and evolve individual contribution to your organisation.
      </p>
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        We therefore ask you to provide feedback based on factual evidence only and we may ask you for further information in some individual performance feedback.
      </p>
    </section>
    <Link
      to="/survey"
    >
      <div
        className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
      >
        <button
          className="StyledButton-sc-323bzc-0 hlJKLl"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          size="large"
          type="button"
        >
          start
        </button>
      </div>
    </Link>
  </div>
</div>
`;

exports[`COMPONENT - Welcome renders correctly without params 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <div
    className="StyledBox-sc-13pk1d4-0 lcYpOF"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 jIghTj"
    >
      Performance Snapshot on RSR Consultant:
    </h1>
    <h2
      className="StyledHeading-sc-1rdh4aw-0 fDHdzw"
    >
      
    </h2>
    <section
      className="sc-bdVaJa gPpPXU"
    >
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        The feedback you provide will be used for individual development and career pathway and forms part of RSR Reward and Recognition scheme.
      </p>
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        Feedback will also be used to enhance and evolve individual contribution to your organisation.
      </p>
      <p
        className="StyledParagraph-tbetod-0 hAoEMb"
        size="large"
      >
        We therefore ask you to provide feedback based on factual evidence only and we may ask you for further information in some individual performance feedback.
      </p>
    </section>
    <Link
      to="/survey"
    >
      <div
        className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
      >
        <button
          className="StyledButton-sc-323bzc-0 hlJKLl"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          size="large"
          type="button"
        >
          start
        </button>
      </div>
    </Link>
  </div>
</div>
`;
